File: 剑指offer全解/42和为S的两个数字.cpp
class Solution {
public:
    vector<int> FindNumbersWithSum(vector<int> array,int sum) {
        vector<int> res;
        if(array.size() < 2) return res;
        int i = 0, j = array.size()-1;
        while(i < j) {
            if(array[i] + array[j] == sum) {
                res.push_back(array[i]);
                res.push_back(array[j]);
                break;
            }
            while(i < j && array[i] + array[j] > sum) j--;
            while(i < j && array[i] + array[j] < sum) i++;
        }
        return res;
    }
};

File: 剑指offer全解/42和为S的两个数字.py
# 第一反应就是python的 in方法
# 因为数组是升序的 所以第一个找到的就是乘积最小的

# -*- coding:utf-8 -*-
class Solution:
    def FindNumbersWithSum(self, array, tsum):
        # write code here
        res = []
        for i in range(len(array)):
            temp = tsum -array[i]
            if tsum - array[i] in array:
                res.append(array[i])
                res.append(tsum-array[i])
                break;
        return res